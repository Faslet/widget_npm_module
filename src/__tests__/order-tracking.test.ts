import { createOrderTracking } from '..';

describe('Order tracking', () => {
  it('generates a valid order tracking tag', () => {
    const orderTracking = createOrderTracking('Faslet Demo');
    orderTracking.withOrderNumber('order-1').withPaymentStatus('paid');

    orderTracking
      .addProduct(
        'product-1',
        'variant-1',
        'Jacket',
        'Red Jacket/S',
        100,
        1,
        'sku-1'
      )
      .addProduct(
        'product-1',
        'variant-2',
        'Jacket',
        'Blue Jacket/S',
        200,
        2,
        'sku-2'
      );

    orderTracking.buildOrderTracking();

    expect(document.head).toMatchSnapshot();
  });

  it('throws an error when shop id is missing', () => {
    expect(() => createOrderTracking('')).toThrow();
  });

  it('throws an error when order number is missing', () => {
    const orderTracking = createOrderTracking('Faslet Demo');
    orderTracking.withPaymentStatus('paid');

    orderTracking
      .addProduct(
        'product-1',
        'variant-1',
        'Jacket',
        'Red Jacket/S',
        100,
        1,
        'sku-1'
      )
      .addProduct(
        'product-1',
        'variant-2',
        'Jacket',
        'Blue Jacket/S',
        200,
        2,
        'sku-2'
      );

    expect(() => orderTracking.buildOrderTracking()).toThrow();
  });

  it('throws an error when products are missing', () => {
    const orderTracking = createOrderTracking('Faslet Demo');
    orderTracking.withOrderNumber('order-1').withPaymentStatus('paid');

    expect(() => orderTracking.buildOrderTracking()).toThrow();
  });
});
