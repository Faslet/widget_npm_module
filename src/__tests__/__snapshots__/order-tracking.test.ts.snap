// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order tracking generates a valid order tracking tag 1`] = `
<head>
  <script
    id="faslet-order-tracking-tag"
    src="https://widget.prod.faslet.net/faslet-orders.js"
    type="text/javascript"
  />
  <script
    defer=""
    id="faslet-gtm-tag"
    src="https://www.googletagmanager.com/gtag/js?id=G-6J8TML143D"
    type="text/javascript"
  />
</head>
`;
